src/main/java/bio/terra/service/snapshotbuilder/SnapshotBuilderService.java

```diff
@@ -190,6 +188,40 @@ public int getRollupCountForCriteriaGroups(
         .get(0);
   }
 
+  private SnapshotBuilderDomainOption getDomainOptionFromSettings(String domainId, UUID datasetId) {
+    SnapshotBuilderSettings snapshotBuilderSettings =
+        snapshotBuilderSettingsDao.getSnapshotBuilderSettingsByDatasetId(datasetId);
+
+    return snapshotBuilderSettings.getDomainOptions().stream()
+        .filter(domainOption -> domainOption.getName().equals(domainId))
+        .findFirst()
+        .orElseThrow(
+            () ->
+                new BadRequestException(
+                    "Invalid domain category is given: %s".formatted(domainId)));
+  }
+
+  private String getDomainId(
```

Comment:
If we are passing in `dataset` here, do we also need to pass in `cloudPlatform`? Did you think about having this method create its own `CloudPlatformWrapper`?

Reply:
I guess my thought was the less places we need to make a CloudPlatformWrapper the better. But if you think its preferable to not pass it in and just get it agin from the dataset, I can do that!